From 6ded2f3339d3ad2ed580e5066d153aef6de57827 Mon Sep 17 00:00:00 2001
From: Xiaofeng Wei <xiaofeng.wei@nxp.com>
Date: Mon, 17 Feb 2025 16:17:11 +0800
Subject: [PATCH 23/30] arm64: dts: add os08a20 device node for i.MX8MP FRDM
 board

Add os08a20 sensor device node for i.MX8MP FRDM board.

Signed-off-by: Xiaofeng Wei <xiaofeng.wei@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../dts/freescale/imx8mp-frdm-os08a20.dts     | 90 +++++++++++++++++++
 2 files changed, 91 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 9f1c78d68d58..bd3ea6588619 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -141,7 +141,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-navq.dtb imx8mp-navq-ov5640-ov5645.dtb imx8mp-n
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-ddr4-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-ndm.dtb
 
-dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb imx8mp-frdm-os08a20.dtb
 
 imx8mp-evk-revb4-dtbs := imx8mp-evk.dtb imx8mp-evk-revb4.dtbo
 imx8mp-evk-revb4-rm67191-dtbs := imx8mp-evk-rm67191.dtb imx8mp-evk-revb4.dtbo
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts
new file mode 100644
index 000000000000..b652ffdc7e80
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-os08a20.dts
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "imx8mp-frdm.dts"
+
+&i2c2 {
+	/delete-node/ov5640_mipi@3c;
+
+	os08a20_0: os08a20_mipi@36 {
+		compatible = "ovti,os08a20_mipi";
+		reg = <0x36>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_mclk>;
+		clocks = <&clk_dummy>;
+		clock-names = "csi_mclk";
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		status = "okay";
+
+		port {
+			os08a20_mipi_0_ep: endpoint {
+				data-lanes = <1 2 3 4>;
+				clock-lanes = <0>;
+				max-pixel-frequency = /bits/ 64 <500000000>;
+				remote-endpoint = <&mipi_csi0_ep>;
+			};
+		};
+
+	};
+};
+
+&i2c3 {
+	/delete-node/ov5640_mipi@3c;
+
+};
+
+&cameradev {
+	status = "okay";
+};
+
+&isi_0 {
+	cap_device {
+		status = "disabled";
+	};
+};
+
+&isi_1 {
+	status = "disabled";
+};
+
+&isp_0 {
+	status = "okay";
+};
+
+&dewarp {
+	status = "okay";
+};
+
+&mipi_csi_0 {
+	status = "okay";
+
+	port {
+		endpoint {
+			remote-endpoint = <&os08a20_mipi_0_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <16>;
+		};
+	};
+};
+
+&mipi_csi_1 {
+	/delete-property/ #address-cells;
+	/delete-property/ #size-cells;
+	/delete-node/port@1;
+};
-- 
2.34.1

